timemachine/ff/handlers/bonded.py

```diff
@@ -139,7 +135,15 @@ def static_parameterize(params, smirks, counts, mol):
 
         scatter_idxs = np.array(scatter_idxs)
 
-        return params[scatter_idxs], np.repeat(torsion_idxs, repeats, axis=0).astype(np.int32)
+        # if no matches found, return arrays that can still be concatenated as expected
+        if len(param_idxs) > 0:
+            assigned_params = params[scatter_idxs]
+            proper_idxs = np.repeat(torsion_idxs, repeats, axis=0).astype(np.int32)
+        else:
+            assigned_params = params[[False] * len(params)]  # empty slice with same dtype, other dimensions
```

Comment:
This confused me at first so I tried to come up with some alternatives: `params[slice(0)]`, `params[0:0]` (not sure if actually clearer but a bit more concise)

Reply:
Either of these alternatives would be an improvement, I think!

Switched to `params[:0]` and retained the comment in https://github.com/proteneer/timemachine/pull/626/commits/98c6a966c7e6e5e25369aeb98ee37ab78ba0052f